packages/expo-updates/android/src/main/java/expo/modules/updates/loader/LoaderTask.kt

```diff
@@ -395,24 +395,15 @@ class LoaderTask(
             }
 
             val embeddedUpdate = EmbeddedManifest.get(context, configuration)!!.updateEntity
-            if (embeddedUpdate == null) {
-              launchUpdate(null)
-              return
-            }
-
-            val launcher = DatabaseLauncher(configuration, directory, fileDownloader, selectionPolicy)
-            val launchableUpdate = launcher.getLaunchableUpdate(database, context)
-            val manifestFilters = ManifestMetadata.getManifestFilters(database, configuration)
 
-            if (!selectionPolicy.shouldLoadRollBackToEmbeddedDirective(updateDirective, embeddedUpdate, launchableUpdate, manifestFilters)) {
```

Comment:
I think the bug here is that this doesn't match what iOS does: use the candidate launcher's update being launched. I'm going to investigate whether just making that change fixes the issue or whether this reordering is necessary too.